src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,104 @@
+package parser;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.HashMap;
+
+import enums.Command;
+import exception.JarvisException;
+
+public class Parser {
+    /**
+     * Parses the input string into a command for jarvis.Jarvis.
+     *
+     * @param input The user input to parse.
+     * @return The parsed command map.
+     * @throws JarvisException If input does not conform to the format.
+     */
+    public static HashMap<String, Object> parseInput(String input) throws JarvisException {
+        HashMap<String, Object> parsedCommand = new HashMap<>();
+        String[] tokens = input.split(" ");
+        parsedCommand.put("command", tokens[0].trim().toUpperCase());
+
+        Command command = Command.valueOf((String) parsedCommand.get("command"));
+        if (command == Command.MARK
```

Comment:
A lot of if-elses here, might be a bit messy. You can consider creating specific exceptions to be thrown, and use a switch case and a single try-catch to handle all the exceptions that could be thrown.